tests/integration/shared-libs/cht-datasource/contact.spec.js

```diff
@@ -0,0 +1,375 @@
+      const placeFreetext = 'clinic';
+
+      it('returns a page of people type contact ids for no limit and cursor passed', async () => {
+        const responsePage = await getIdsPage(Qualifier.byContactType(personType));
+        const responsePeople = responsePage.data;
+        const responseCursor = responsePage.cursor;
+
+        expect(responsePeople).excludingEvery([ '_rev', 'reported_date' ]).to.deep.equalInAnyOrder(expectedPeopleIds);
+        expect(responseCursor).to.be.equal(null);
+      });
+
+      it('returns a page of place type contact for no limit and cursor passed', async () => {
+        const responsePage = await getIdsPage(Qualifier.byContactType(placeType));
+        const responsePlaces = responsePage.data;
+        const responseCursor = responsePage.cursor;
+
+        expect(responsePlaces).excludingEvery([ '_rev', 'reported_date' ])
+          .to.deep.equalInAnyOrder(expectedPlacesIds);
+        expect(responseCursor).to.be.equal(null);
+      });
+
+      it('returns a page of contact ids for freetext with no limit and cursor passed', async () => {
+        const expectedContactIds = [ contact0._id, contact1._id, contact2._id ];
+        const responsePage = await getIdsPage(Qualifier.byFreetext(freetext));
+        const responsePeople = responsePage.data;
+        const responseCursor = responsePage.cursor;
+
+        expect(responsePeople).excludingEvery([ '_rev', 'reported_date' ]).to.deep.equalInAnyOrder(expectedContactIds);
+        expect(responseCursor).to.be.equal(null);
+      });
+
+      it('returns a page of people type contact ids and freetext for no limit and cursor passed', async () => {
+        const responsePage = await getIdsPage({
+          ...Qualifier.byContactType(personType), ...Qualifier.byFreetext(freetext),
+        });
+        const expectedContactIds = [ contact0._id, contact1._id, contact2._id ];
+        const responsePeople = responsePage.data;
+        const responseCursor = responsePage.cursor;
+
+        expect(responsePeople).excludingEvery([ '_rev', 'reported_date' ]).to.deep.equalInAnyOrder(expectedContactIds);
+        expect(responseCursor).to.be.equal(null);
+      });
+
+      it('returns a page of place type contact with freetext for no limit and cursor passed', async () => {
+        const freetext = 'clinic';
+        const responsePage = await getIdsPage({
+          ...Qualifier.byContactType(placeType), ...Qualifier.byFreetext(freetext)
+        });
+        const responsePlaces = responsePage.data;
+        const responseCursor = responsePage.cursor;
+        const expectedContactIds = [ place0._id, clinic1._id, clinic2._id ];
+
+        expect(responsePlaces).excludingEvery([ '_rev', 'reported_date' ])
+          .to.deep.equalInAnyOrder(expectedContactIds);
+        expect(responseCursor).to.be.equal(null);
+      });
+
+      it('returns a page of people type contact ids ' +
+      'when stringified limit and null cursor is passed', async () => {
```

Comment:
nitpick: IMHO this does not need to be an integration test.  This functionality is adequately covered by unit tests.